clients/client/src/main/java/org/projectnessie/client/http/HttpClient.java

```diff
@@ -137,11 +164,37 @@ public HttpClient build() {
               .connectionTimeoutMillis(connectionTimeoutMillis)
               .isDisableCompression(disableCompression)
               .sslContext(sslContext)
+              .sslParameters(sslParameters)
               .addAllRequestFilters(requestFilters)
               .addAllResponseFilters(responseFilters)
+              .isHttp11Only(!http2Upgrade)
+              .followRedirects(followRedirects)
+              .forceUrlConnectionClient(forceUrlConnectionClient)
               .build();
 
-      return new UrlConnectionClient(config);
+      return ImplSwitch.FACTORY.apply(config);
+    }
+
+    static class ImplSwitch {
+      static final Function<HttpRuntimeConfig, HttpClient> FACTORY;
+
+      static {
+        Function<HttpRuntimeConfig, HttpClient> factory;
+        try {
+          Class.forName("java.net.http.HttpClient");
+          factory =
+              config ->
+                  // Need the system property for tests, "normal" users can use standard
+                  // configuration options.
+                  Boolean.getBoolean("nessie.client.force-url-connection-client")
```

Comment:
nit: I see tests run under Java8 to exercise `UrlConnectionClient`, but I did not find any references to this system property :thinking:

Reply:
Yep. It's only there as a "last resort fallback".

Reply:
ok